File: app/src/main/java/com/niccher/mpesa_analyzer/frags/Frag_Profile.java
package com.niccher.mpesa_analyzer.frags;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.appcompat.app.ActionBar;
import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;

import com.niccher.mpesa_analyzer.R;

public class Frag_Profile extends Fragment {

    AppCompatActivity activity;

    public Frag_Profile() {
        // Required empty public constructor
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        activity = (AppCompatActivity) getActivity();
        ActionBar supportActionBar = activity.getSupportActionBar();
        if (supportActionBar != null) {
            supportActionBar.setTitle("Profile");
            supportActionBar.setDisplayHomeAsUpEnabled(false);
        }
        setHasOptionsMenu(true);
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.frag_profile, container, false);
    }
}

File: app/src/main/java/com/niccher/mpesa_analyzer_app/fragments/Profile/Frag_Profile.kt
package com.niccher.mpesa_analyzer_app.fragments.Profile

import android.content.Context
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.fragment.app.Fragment
import com.niccher.mpesa_analyzer_app.R
import com.niccher.mpesa_analyzer_app.konstants.Konstants

class Frag_Profile : Fragment() {

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View = inflater.inflate(R.layout.frag_profile, container, false)

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        // Load user data from SharedPreferences (same prefs used by auth)
        val prefs = requireContext().getSharedPreferences(
            Konstants.SHARED_AUTH_LOGIN, Context.MODE_PRIVATE
        )

        val username = prefs.getString("userName", "User")?.replaceFirstChar { it.uppercase() } ?: "User"
        val email = prefs.getString("userEmail", "") ?: ""
        val memberSince = prefs.getString("time", "") ?: ""

        view.findViewById<TextView>(R.id.tv_username).text = username
        if (email.isNotBlank() && email != "nullable") {
            view.findViewById<TextView>(R.id.tv_email).text = email
        }
        if (memberSince.isNotBlank() && memberSince != "nullable") {
            view.findViewById<TextView>(R.id.tv_member_since).text = "Logged in at $memberSince"
        }
    }
}
